gridcomps/History3G/HistoryCollectionGridComp.F90

```diff
@@ -98,8 +108,21 @@ subroutine run(gridcomp, importState, exportState, clock, rc)
       integer, intent(out)  :: rc
 
       integer :: status
+      type(HistoryCollectionGridComp), pointer :: collection_gridcomp
+      character(*), parameter :: PRIVATE_STATE = "HistoryCollectionGridComp"
+      logical :: time_to_write, run_collection
+      type(ESMF_Time) :: current_time
+    
+      call ESMF_ClockGet(clock, currTime=current_time, _RC) 
+      _GET_NAMED_PRIVATE_STATE(gridcomp, HistoryCollectionGridComp, PRIVATE_STATE, collection_gridcomp)
+      time_to_write = ESMF_AlarmIsRinging(collection_gridcomp%write_alarm, _RC)
```

Comment:
There is a subtle issue lurking here.  Namely,  I had envisioned that the outer/generic gridcomp would manage the basic process of when should a run method be called.

At the same time, we expect this information to be processed a bit differently in HistoryCollection than it will in other gridcomps.

The middle path might be that History adds a `mapl` section to the config that it passes down to HistoryCollection and that lets the generic layer intercede appropriately.

But we can keep your implementation here while we debate it in the more general context.

Reply:
@tclune, I had no idea that was your vision

Reply:
> @tclune, I had no idea that was your vision
@tclune
And there still this issue related to when things run.
https://github.com/GEOS-ESM/MAPL/issues/2792

Reply:
Not a vision baked in stone (too mix multiple metaphors), but it seems like it would be good for the generic layer to handle this and avoid repetitive code in each gridcomp.   Esp. since the gridcomp might get it wrong which could be a subtle bug.